File: Client/src/app/models/full-rental.model.ts
import { CarModel } from './car.model';
import { BranchModel } from './branch.model';
import { UserModel } from './user.model';
import { RentalModel } from "./rental.model";

export class FullRentalModel {
    constructor(
        public rentalModel?: RentalModel,
        public userModel?: UserModel,
        public branchModelPickUp?: BranchModel,
        public branchModelReturn?: BranchModel,
        public carModel?: CarModel
    ){}
}

File: Server/02 - Business Model Layer/FullRentalModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CarRental
{
    public class FullRentalModel
    {
        public RentalModel rentalModel { get; set; }
        public UserModel userModel { get; set; }
        public BranchModel branchModelPickUp { get; set; }
        public BranchModel branchModelReturn { get; set; }
        public CarModel carModel { get; set; }
    }
}
